hardware_interface/test/test_resource_manager.cpp

```diff
@@ -350,3 +386,803 @@ TEST_F(TestResourceManager, custom_prepare_perform_switch)
   EXPECT_FALSE(rm.perform_command_mode_switch(empty_keys, empty_keys));
   EXPECT_FALSE(rm.perform_command_mode_switch(empty_keys, legal_keys_position));
 }
+
+TEST_F(TestResourceManager, resource_status)
+{
+  hardware_interface::ResourceManager rm(ros2_control_test_assets::minimal_robot_urdf);
+
+  auto status_map = rm.get_components_status();
+
+  // name
+  EXPECT_EQ(status_map[TEST_ACTUATOR_HARDWARE_NAME].name, TEST_ACTUATOR_HARDWARE_NAME);
+  EXPECT_EQ(status_map[TEST_SENSOR_HARDWARE_NAME].name, TEST_SENSOR_HARDWARE_NAME);
+  EXPECT_EQ(status_map[TEST_SYSTEM_HARDWARE_NAME].name, TEST_SYSTEM_HARDWARE_NAME);
+  // type
+  EXPECT_EQ(status_map[TEST_ACTUATOR_HARDWARE_NAME].type, TEST_ACTUATOR_HARDWARE_TYPE);
+  EXPECT_EQ(status_map[TEST_SENSOR_HARDWARE_NAME].type, TEST_SENSOR_HARDWARE_TYPE);
+  EXPECT_EQ(status_map[TEST_SYSTEM_HARDWARE_NAME].type, TEST_SYSTEM_HARDWARE_TYPE);
+  // class_type
+  EXPECT_EQ(status_map[TEST_ACTUATOR_HARDWARE_NAME].class_type, TEST_ACTUATOR_HARDWARE_CLASS_TYPE);
+  EXPECT_EQ(status_map[TEST_SENSOR_HARDWARE_NAME].class_type, TEST_SENSOR_HARDWARE_CLASS_TYPE);
+  EXPECT_EQ(status_map[TEST_SYSTEM_HARDWARE_NAME].class_type, TEST_SYSTEM_HARDWARE_CLASS_TYPE);
+  // state
+  EXPECT_EQ(
+    status_map[TEST_ACTUATOR_HARDWARE_NAME].state.id(),
+    lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED);
+  EXPECT_EQ(
+    status_map[TEST_ACTUATOR_HARDWARE_NAME].state.label(),
+    hardware_interface::lifecycle_state_names::UNCONFIGURED);
+  EXPECT_EQ(
+    status_map[TEST_SENSOR_HARDWARE_NAME].state.id(),
+    lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED);
+  EXPECT_EQ(
+    status_map[TEST_SENSOR_HARDWARE_NAME].state.label(),
+    hardware_interface::lifecycle_state_names::UNCONFIGURED);
+  EXPECT_EQ(
+    status_map[TEST_SYSTEM_HARDWARE_NAME].state.id(),
+    lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED);
+  EXPECT_EQ(
+    status_map[TEST_SYSTEM_HARDWARE_NAME].state.label(),
+    hardware_interface::lifecycle_state_names::UNCONFIGURED);
+
+  auto check_interfaces = [](
+                            const std::vector<std::string> & registered_interfaces,
+                            const std::vector<const char *> & interface_names) {
+    for (const auto & interface : interface_names)
+    {
+      auto it = std::find(registered_interfaces.begin(), registered_interfaces.end(), interface);
+      EXPECT_NE(it, registered_interfaces.end());
+    }
+  };
+
+  check_interfaces(
+    status_map[TEST_ACTUATOR_HARDWARE_NAME].command_interfaces,
+    TEST_ACTUATOR_HARDWARE_COMMAND_INTERFACES);
+  EXPECT_TRUE(status_map[TEST_SENSOR_HARDWARE_NAME].command_interfaces.empty());
+  check_interfaces(
+    status_map[TEST_SYSTEM_HARDWARE_NAME].command_interfaces,
+    TEST_SYSTEM_HARDWARE_COMMAND_INTERFACES);
+
+  check_interfaces(
+    status_map[TEST_ACTUATOR_HARDWARE_NAME].state_interfaces,
+    TEST_ACTUATOR_HARDWARE_STATE_INTERFACES);
+  // TODO(destogl) we allow unlisted interfaces.... Is this OK? (@bmagyar)
```

Comment:
what does unlisted mean? not listed in the URDF? I think that should be ok

Reply:
Exactly. They are not in the URDF, but added in the hardware directly.